packages/ui-components/src/components/time-picker-calendar/time-picker-calendar.tsx

```diff
@@ -0,0 +1,302 @@
+
+	@State() month: number = getDateMonth(this.initialDate ?? new Date());
+	@State() year: number = getDateYear(this.initialDate ?? new Date());
+	@State() hoveredDay: number;
+
+	@Watch('selectedDates')
+	validateSelectedDates(newSelectedDates: string[] | undefined) {
+		if (newSelectedDates !== undefined && !areDatesValid(newSelectedDates)) {
+			throw new Error('Selected date should be an array with valid dates');
+		}
+	}
+
+	@Watch('initialDate')
+	validateInitialDate(newInitialDate: string | undefined) {
+		if (newInitialDate !== undefined && !isDateValid(newInitialDate)) {
+			throw new Error('Initial date should be a valid date');
+		} else {
+			this.month = getDateMonth(newInitialDate ?? new Date());
+			this.year = getDateYear(newInitialDate ?? new Date());
+		}
+	}
+
+	@Watch('disabledDates')
+	validateDisabledDates(newDisabledDates: string[] | undefined) {
+		if (newDisabledDates !== undefined && !areDatesValid(newDisabledDates)) {
+			throw new Error('Disabled dates should be an array with valid dates');
+		}
+	}
+
+	componentWillLoad() {
+		this.validateSelectedDates(this.selectedDates);
+		this.validateInitialDate(this.initialDate);
+		this.validateDisabledDates(this.disabledDates);
+	}
+
+	public onClickPreviousMonth = (event: MouseEvent) => {
+		if (this.month === 1) {
+			this.year = this.year - 1;
+			this.month = 12;
+
+			this.changeMonth.emit({ event: event, payload: this.month });
+			this.changeYear.emit({ event: event, payload: this.year });
+
+			return;
+		}
+
+		this.month = this.month - 1;
+		this.changeMonth.emit({ event: event, payload: this.month });
+	};
+
+	public onClickNextMonth = (event: MouseEvent) => {
+		if (this.month === 12) {
+			this.year = this.year + 1;
+			this.month = 1;
+
+			this.changeMonth.emit({ event: event, payload: this.month });
+			this.changeYear.emit({ event: event, payload: this.year });
+
+			return;
+		}
+
+		this.month = this.month + 1;
+		this.changeMonth.emit({ event: event, payload: this.month });
+	};
+
+	public isDayDisabled = (day: number): boolean => {
+		const dayMoment = fromDateFields(day, this.month, this.year);
+
+		if (isDateInArray(dayMoment, this.disabledDates)) {
+			return true;
+		}
+
+		if (this.minDate && isDateBefore(dayMoment, this.minDate)) {
+			return true;
+		}
+
+		if (this.maxDate && isDateAfter(dayMoment, this.maxDate)) {
+			return true;
+		}
+
+		return false;
+	};
+
+	public onClickDay = (day: number, event: MouseEvent): void => {
+		const clickedDateMoment = fromDateFields(day, this.month, this.year);
+		this.clickDate.emit({ event, payload: formatDate(clickedDateMoment) });
+	};
+
+	public isDayActive = (day: number): boolean => {
+		const date = fromDateFields(day, this.month, this.year);
+
+		return isDateInArray(date, this.selectedDates);
+	};
+
+	public onMouseEnter = (day: number): void => {
+		this.hoveredDay = day;
+		const date = fromDateFields(day, this.month, this.year).format(DATE_FORMAT);
+		this.hoveredDateChange.emit(date);
```

Comment:
It is. We are suppose to show the hover state in both calendars. Without sending the the hovered date to the parent component, we cannot know if we are hovering a date on another calendar and keep the hover state in the current calendar. We cannot remove this otherwise we will have hover only in the calendar we are hovering instead of displaying in both calendars.